### PR TITLE
Add 10-percent heuristic to httpCache.isStale()

### PR DESCRIPTION
**Pull request by noahadams**

### CODE REVIEW

File: api/combo.js [hunk 1]
```diff
@@ -183,17 +183,35 @@ var ccDirectives = /^\s*(public|private|no-cache|no-store)\s*$/
...
+                lastModified = Date.parse(lastModified);
+                modifiedAge = date - lastModified;
+                age = now - date;
+                // If the age is less than 10% of the time between the last 
+                // modification and the response, and the age is less than a 
+                // day, then it is not stale
+                if ((age < 0.1 * modifiedAge) && (age < ONE_DAY_IN_MS)) {
+                    return false;
+                }
+            }
```
Comment (rrjamie) [inline api/combo.js]: Are we sure we want to use this heuristic even if we have a `cache-control` header?

File: api/combo.js [hunk 2]
```diff
@@ -183,17 +183,35 @@ var ccDirectives = /^\s*(public|private|no-cache|no-store)\s*$/
...
-              // Fresh if less than 10 minutes old
+            // Fresh if less than 10 minutes old
             if (date && (now < date + 600 * 1000)) {
                return false;
             }
 
+            // Fresh if less than 10% of difference between date and 
+            // last-modified old, up to a day
+            if (lastModified && date) {
+                lastModified = Date.parse(lastModified);
```
Comment (rrjamie) [inline api/combo.js]: are we reasonably sure that lastModified won't be some junk data?

File: api/combo.js (hunk 1 above)
Comment (noahadams) [inline api/combo.js]: you're right here, the control flow on this needs some work

**Comment by noahadams** [comment]
@rrjamie flow control rejiggered for correctness and test case modified to illustrate it.

File: tests/index.html [hunk 3]
```diff
@@ -523,6 +549,9 @@ <h2 id="qunit-userAgent"></h2>
             ok(isStale(resources['invalid-expires']), 'Invalid expires is stale');
             // Test our ten minute minimum
             ok(!isStale(resources['new-no-cache-headers']), 'No cache headers but less than 10 minutes old');
+            // Test ten percent heuristic
+            debugger;
```
Comment (rrjamie) [inline tests/index.html]: what the hell is this shit right here

Comment (noahadams) [inline tests/index.html]: My shame Ryan, it's my shame

**Comment by noahadams** [comment]
verbal :+1: from @rrjamie

**Pull request merged**